api/v1beta1/metal3machine_types.go

```diff
@@ -33,6 +33,81 @@ const (
 	LiveIsoDiskFormat    = "live-iso"
 )
 
+// Metal3Machine's Ready condition and corresponding reasons that will be used in v1Beta2 API version.
+const (
+	// Metal3MachineReadyV1Beta2Condition is true if the Metal3Machine's deletionTimestamp is not set, Metal3Machine's
+	// BareMetalHostProvisioned is true.
+	Metal3MachineReadyV1Beta2Condition = clusterv1beta1.ReadyV1Beta2Condition
+
+	// Metal3MachineReadyV1Beta2Reason surfaces when the Metal3Machine readiness criteria is met.
+	Metal3MachineReadyV1Beta2Reason = clusterv1beta1.ReadyV1Beta2Reason
+
+	// Metal3MachineNotReadyV1Beta2Reason surfaces when the Metal3Machine readiness criteria is not met.
+	Metal3MachineNotReadyV1Beta2Reason = clusterv1beta1.NotReadyV1Beta2Reason
+
+	// Metal3MachineReadyUnknownV1Beta2Reason surfaces when at least one Metal3Machine readiness criteria is unknown
+	// and no Metal3Machine readiness criteria is not met.
+	Metal3MachineReadyUnknownV1Beta2Reason = clusterv1beta1.ReadyUnknownV1Beta2Reason
+)
+
+// Metal3Machine condition and corresponding reasons that will be used in v1Beta2 API version.
+const (
+	// AssociateBareMetalHostV1Beta2Condition documents the status of the association of Metal3Machine with a BareMetalHost.
+	AssociateBareMetalHostV1Beta2Condition = "AssociateBareMetalHost"
+
+	// AssociateBareMetalHostFailedV1Beta2Reason documents any errors while associating Metal3Machine with a BareMetalHost.
+	AssociateBareMetalHostFailedV1Beta2Reason = "AssociateBareMetalHostFailed"
+
+	// AssociateBareMetalHostSuccessV1Beta2Reason surfaces when the Metal3Machine is successfully associated with a BareMetalHost.
+	AssociateBareMetalHostSuccessV1Beta2Reason = "AssociateBareMetalHostSuccess"
+
+	// WaitingForClusterInfrastructureReasonReadyV1Beta2Reason used when waiting for cluster
+	// infrastructure to be ready before proceeding.
+	WaitingForClusterInfrastructureReasonReadyV1Beta2Reason = clusterv1beta1.WaitingForClusterInfrastructureReadyV1Beta2Reason
+
+	// WaitingForBootstrapDataV1Beta2Reason used when waiting for bootstrap to be ready before proceeding.
+	WaitingForBootstrapDataV1Beta2Reason = clusterv1beta1.WaitingForBootstrapDataV1Beta2Reason
+
+	// WaitingForMetal3MachineOwnerRefV1Beta2Reason is used when Metal3Machine is waiting for OwnerReference to be
+	// set before proceeding.
+	WaitingForMetal3MachineOwnerRefV1Beta2Reason = "WaitingForMetal3MachineOwnerRef"
+
+	// WaitingforMetal3ClusterReason is used when Metal3Machine is waiting for Metal3Cluster.
+	WaitingforMetal3ClusterV1Beta2Reason = "WaitingforMetal3Cluster"
```

Comment:
How is this different from WaitingForClusterInfrastructureReadyV1Beta2Reason? Is that for the CAPI cluster and this is for the Metal3Cluster? I'm getting confused by the "ClusterInfrastructure". It sounds like the Metal3Cluster to me

Reply:
We set `WaitingForClusterInfrastructureReadyV1Beta2Reason` when checking the `InfrastructureReady` condition on the CAPI Cluster.
We set `WaitingForMetal3ClusterV1Beta2Reason` when attempting to retrieve the `Metal3Cluster`.

In the `metal3machine_controller`, these are set in order: first `WaitingForClusterInfrastructureReadyV1Beta2Reason`, then `WaitingForMetal3ClusterV1Beta2Reason`.

Since a CAPI Cluster marked as InfrastructureReady implies that the corresponding Metal3Cluster is also ready and retrievable, it’s unclear whether we actually need the WaitingForMetal3ClusterV1Beta2Reason.